arcade/sprite_list/collision.py

```diff
@@ -93,32 +93,36 @@ def _check_for_collision(sprite1: SpriteType, sprite2: SpriteType) -> bool:
     :returns: True if sprites overlap.
     :rtype: bool
     """
+    
+    #NOTE: for speed becuase attribute look ups are slow.
     sprite1_position = sprite1._position
     sprite1_width = sprite1._width
     sprite1_height = sprite1._height
     sprite2_position = sprite2._position
     sprite2_width = sprite2._width
     sprite2_height = sprite2._height
+
     radius_sum = (
         (sprite1_width if sprite1_width > sprite1_height else sprite1_height)
         + (sprite2_width if sprite2_width > sprite2_height else sprite2_height)
```

Comment:
@cspotcode what's the overhead of this vs `max` on recent Python versions?

Reply:
I forget, but I do remember doing my benchmarking on 3.11.  We could bench again and find out.   `min()` is better than `math.min()` because the latter has an extra dictionary lookup.

Reply:
I’ll mark this as resolved for now. We can evaluate `min` and `max` performance in it’s own issue.